docs/learn/tutorials/versioned/samza-async-user-guide.md

```diff
@@ -107,16 +107,96 @@ task.callback.timeout.ms=5000
+
+{% highlight jproperties %}
+# Timeout for the future to complete. When the timeout elapses, the framework will throw a TaskCallbackTimeoutException and shut down the container.
+task.callback.timeout.ms
+{% endhighlight %}
+
+For remote IO libraries that don't support Futures/Promises and use callbacks, it should be straightforward to adapt the callbacks to Future pattern. The following code snippet illustrates it
+
+{% highlight java %}
+
+  public CompletionStage<Collection<StandardizedWikipediaFeedEvent>> apply(WikipediaFeedEvent wikipediaFeedEvent) {
+    Request<StandardizerRequest> standardizedRequest = buildStandardizedRequest(wikipediaFeedEvent);
+    CompletableFuture<Collection<StandardizedWikipediaFeedEvent>> standardizedFuture = new CompletableFuture<>();
+    client.async().get(standardizedRequest, new InvocationCallback<Response>() {
+          @Override
+          public void completed(ResStandardizerResponseponse response) {
+            standardizedFuture.complete(extractStandardizedWikipediaFeedEvent(response));
+          }
+
+          @Override
+          public void failed(Throwable throwable) {
+            standardizedFuture.completeExceptionally(throwable);
+          }
+        });
+  }
+{% endhighlight %}
+
 ### Out-of-order Process
 
-In both cases above, Samza supports in-order process by default. Further parallelism is also supported by allowing a task to process multiple outstanding messages in parallel. The following config allows one task to process at most 4 outstanding messages in parallel at a time: 
+In all cases above, Samza supports in-order process by default. Further parallelism is also supported by allowing a task to process multiple outstanding messages in parallel. The following config allows one task to process at most 4 outstanding messages in parallel at a time:
 
 {% highlight jproperties %}
 # Max number of outstanding messages being processed per task at a time, applicable to both StreamTask and AsyncStreamTask.
 task.max.concurrency=4
 {% endhighlight %}
 
-**NOTE:** In case of AsyncStreamTask, processAsync() is still invoked in the order of the message arrivals, but the completion can go out of order. In case of StreamTask with multithreading, process() can run out-of-order since they are dispatched to a thread pool. This option should **NOT** be used when strict ordering of the output is required.
+**NOTE:** In case of AsyncStreamTask, processAsync() is still invoked in the order of the message arrivals, but the completion can go out of order. In case of StreamTask and StreamApplication with concurrency, process() can run out-of-order since they are dispatched to a thread pool. This option should **NOT** be used when strict ordering of the output is required.
```

Comment:
s/can go/can happen.

s/StreamApplication with concurrency/High Level API applications with task.max.concurrency > 1,

s/process() can run/delivery can be.

"dispatched to a thread pool" isn't relevant here.